calibrators/mapping_based_calibrator/README.md

```diff
@@ -0,0 +1,238 @@
+
+##### Data selection
+
+To refine our calibration process, the tool first temporarily pairs `mapping lidar` keyframes with their closest `calibration lidar` frame, and then selects the final calibration pairs (`mapping lidar` keyframes and `calibration lidar` frame) based on the criteria listed below.
+
+- The pairs have low time difference and low interpolation error (such as speed, estimated acceleration, etc.) between the keyframe and the `calibration lidar` frame.
+- The pointlcoud in `calibration lidar` frame has enough features for calibration, which the tool filters out the pointcloud that has low variance in the z-axis that is mostly a plane.
+
+##### Data preprocessing
+
+For the `calibration lidar` pointcloud to have more points to apply the registration algorithm, the calibrator would need a high-resolution pointcloud from the `mapping lidar`. Therefore, we augment the keyframe's pointcloud with all the frames' pointcloud near it. As the resulting pointcloud has redundant information, we downsample the pointcloud to quite a fine-grained resolution, which is called augmented pointcloud.
+
+#### Step 3: Estimate transformations
+
+Once the data is prepared for calibration, the tool applies registration algorithms, such as NDT and GICP, to estimate the transformation between the pointcloud from the `calibration lidar` and the augmented pointcloud from the `mapping lidar`.
+
+### base-lidar calibration
+
+Instead of calibrating the transformations between lidars, we can also utilize the map generated by the `mapping lidar` to calibrate the transformation between the `mapping lidar` and the `base_link`.
+
+#### Step 1: Construct a map
+
+The first step of base-lidar calibration would be the same as [Step 1](#step-1-construct-a-map) in `lidar-lidar calibration`.
+
+#### Step 2: Extract ground plane from the pointcloud
+
+After constructing the map, we could get the augmented pointcloud from `mapping lidar`, which would be the same process as [Step 2](#step-2-prepare-calibration-data). Afterward, the tool utilizes PCA and a segmentation algorithm for extracting the ground plane from the augmented pointcloud.
+
+#### Step 3: Estimate transformation
+
+To estimate the transformation between the `mapping lidar` and the `baselink`, the tool needs to calculate the transformation between the lidar and the ground pose, as well as the transformation between the ground pose and the baselink.
+
+The transformation between the lidar and the ground pose is calculated by utilizing the normal vector and a point on the ground plane. To estimate the transformation between the ground pose and the baselink, the tool first determines the initial ground-pose-to-baselink using the initial lidar-to-baselink and lidar-to-ground-pose transformations. Then, the tool projects this initial ground-pose-to-baselink transformation onto the xy plane to estimate the transformation between the ground pose and the baselink.
```

Comment:
baselink

Reply:
fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/9d68e7a7bb19858191090e5dc7658d92883fc208